ipa-core/src/protocol/basics/reveal.rs

```diff
@@ -119,53 +124,14 @@ impl<C: Context, V: SharedValue + Vectorizable<N>, const N: usize> Reveal<C, N>
 /// to both helpers (right and left) and upon receiving 2 shares from peers it validates that they
 /// indeed match.
 #[cfg(feature = "descriptive-gate")]
-#[async_trait]
 impl<'a, F: ExtendableField> Reveal<UpgradedMaliciousContext<'a, F>, 1> for MaliciousReplicated<F> {
     type Output = <F as Vectorizable<1>>::Array;
 
-    async fn reveal<'fut>(
-        &self,
-        ctx: UpgradedMaliciousContext<'a, F>,
-        record_id: RecordId,
-    ) -> Result<<F as Vectorizable<1>>::Array, Error>
-    where
-        UpgradedMaliciousContext<'a, F>: 'fut,
-    {
-        use futures::future::try_join;
-
-        use crate::secret_sharing::replicated::malicious::ThisCodeIsAuthorizedToDowngradeFromMalicious;
-
-        let (left, right) = self.x().access_without_downgrade().as_tuple();
-        let left_sender = ctx.send_channel(ctx.role().peer(Direction::Left));
-        let left_receiver = ctx.recv_channel::<F>(ctx.role().peer(Direction::Left));
-        let right_sender = ctx.send_channel(ctx.role().peer(Direction::Right));
-        let right_receiver = ctx.recv_channel::<F>(ctx.role().peer(Direction::Right));
-
-        // Send share to helpers to the right and left
-        try_join(
-            left_sender.send(record_id, right),
-            right_sender.send(record_id, left),
-        )
-        .await?;
-
-        let (share_from_left, share_from_right) = try_join(
-            left_receiver.receive(record_id),
-            right_receiver.receive(record_id),
-        )
-        .await?;
-
-        if share_from_left == share_from_right {
-            Ok((left + right + share_from_left).into_array())
-        } else {
-            Err(Error::MaliciousRevealFailed)
-        }
-    }
-
-    async fn partial_reveal<'fut>(
-        &self,
+    async fn generic_reveal<'fut>(
+        &'fut self,
         ctx: UpgradedMaliciousContext<'a, F>,
         record_id: RecordId,
-        left_out: Role,
+        left_out: Option<Role>,
```

Comment:
An observation, unrelated to this change:

"left" out is a poor name in this context because of the use of "left" and "right" as subjective labels for helpers.  "skipped" or "excluded" perhaps?